### PR TITLE
New version: Ryochan7.DS4Windows version 3.3.0

### LINKED ISSUE #130665: [Package Request]: DS4Windows
### How can we help?

I would like someone else to build the manifest.

### Please read and ensure the following

- [X] The installer meets the above requirements

### Please provide the following information

Download Page Url: https://github.com/Ryochan7/DS4Windows/releases
Publisher: Ryochan7
Package Name: DS4Windows
Description: DS4Windows is a utility designed to bring the Sony DualShock 4 game controllers into popular PC games that don't specifically support it
Package Version: 3.2.21
Installer URL: https://github.com/Ryochan7/DS4Windows/releases/download/v3.2.21/DS4Windows_installer_x64_v3.2.21.exe

### PR DESCRIPTION
**Pull request by Dragon1573**
## Description

- Resolve #130665

## Checks

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/130690)

### CODE REVIEW

File: manifests/r/Ryochan7/DS4Windows/3.3.0/Ryochan7.DS4Windows.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+PackageIdentifier: Ryochan7.DS4Windows
+PackageVersion: 3.3.0
+InstallerType: nullsoft
+Scope: machine
+Dependencies:
+  PackageDependencies:
+  - PackageIdentifier: Microsoft.VCRedist.2015+.x64
+  - PackageIdentifier: Microsoft.DotNet.DesktopRuntime.8
+  - PackageIdentifier: ViGEm.ViGEmBus
+  - PackageIdentifier: ViGEm.HidHide
```
Comment (mdanish-kh) [inline manifests/r/Ryochan7/DS4Windows/3.3.0/Ryochan7.DS4Windows.installer.yaml:13]:
```suggestion
```

Based on https://github.com/Ryochan7/DS4Windows#requirements, I don't think these two dependencies are necessary. Can you remove them and then we can check if the package passes the validation pipelines or not. If it doesn't, we can add them back in. If it does, we should also remove these dependencies in the previous version.

File: manifests/r/Ryochan7/DS4Windows/3.3.0/Ryochan7.DS4Windows.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.5.7.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.5.0.schema.json
+
+PackageIdentifier: Ryochan7.DS4Windows
+PackageVersion: 3.3.0
+PackageLocale: en-US
+Publisher: Travis Nickles
```
Comment (mdanish-kh) [inline manifests/r/Ryochan7/DS4Windows/3.3.0/Ryochan7.DS4Windows.locale.en-US.yaml:7]:
```suggestion
Publisher: Ryochan7
```

File: manifests/r/Ryochan7/DS4Windows/3.3.0/Ryochan7.DS4Windows.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+Installers:
+- Architecture: x64
+  InstallerUrl: https://github.com/Ryochan7/DS4Windows/releases/download/v3.3.0/DS4Windows_installer_x64_v3.3.0.exe
+  InstallerSha256: 3D58356CF941FB739F24AF900A5AE9E9BFB3A18EC1986008D7D447F9DCB77BD2
```
Comment (mdanish-kh) [inline manifests/r/Ryochan7/DS4Windows/3.3.0/Ryochan7.DS4Windows.installer.yaml:17]:
```suggestion
  InstallerSha256: 3D58356CF941FB739F24AF900A5AE9E9BFB3A18EC1986008D7D447F9DCB77BD2
  ProductCode: DS4Windows
```

Additional metadata.

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

File: manifests/r/Ryochan7/DS4Windows/3.3.0/Ryochan7.DS4Windows.installer.yaml (hunk 1 above)
Comment (Dragon1573) [reply manifests/r/Ryochan7/DS4Windows/3.3.0/Ryochan7.DS4Windows.installer.yaml]: Nope. Manually validated in Windows Sandbox, the DS4Windows installer will automatically launch wizard of `ViGEm.ViGemBus` and `ViGEm.HidHide`. This behaviour can not be bypassed during installation.

As the above 2 packages are already presented in the repository and can be installed unattendedly by `winget.exe`, we HAVE TO add them as package dependencies, preventing their installation wizard from launching by DS4Windows'.

image

image

**Comment by Dragon1573** [comment]
> [!IMPORTANT]
>
> Commit https://github.com/microsoft/winget-pkgs/pull/130690/commits/a02235141d2a09505fc41cf1ae609591934d1937 should be reverted. See https://github.com/microsoft/winget-pkgs/pull/130690#discussion_r1429618531 for more details.

**Review by mdanish-kh — APPROVED** [review]

**Pull request merged**